src/czar/CzarRegistry.cc

```diff
@@ -0,0 +1,191 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "czar/CzarRegistry.h"
+
+// System headers
+#include <stdexcept>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+// Qserv headers
+#include "cconfig/CzarConfig.h"
+#include "czar/CzarChunkMap.h"
+#include "czar/Czar.h"
+#include "http/Client.h"
+#include "http/Method.h"
+#include "util/common.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+using namespace std;
+using namespace nlohmann;
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.czar.CzarRegistry");
+}  // namespace
+
+namespace lsst::qserv::czar {
+
+CzarRegistry::CzarRegistry(std::shared_ptr<cconfig::CzarConfig> const& czarConfig) : _czarConfig(czarConfig) {
+    // Begin periodically updating worker's status in the Replication System's registry
+    // in the detached thread. This will continue before the application gets terminated.
+    thread _registryUpdateThread(&CzarRegistry::_registryUpdateLoop, this);
+    _czarHeartbeatThrd = move(_registryUpdateThread);
```

Comment:
The name of a local variable `_registryUpdateThread` doesn't comply with the LSST C++ style guide. It should not start with `_`. Besides, you don't need to introduce such a name as you can make the temporary and move in a single line:
```
 _czarHeartbeatThrd = move(thread((&CzarRegistry::_registryUpdateLoop, this)));
 ```

Reply:
It is a private member variable so _loop and join can be used and other options can be available later for deciding what to do if communications have problems.